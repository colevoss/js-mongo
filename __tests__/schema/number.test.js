const Schema = require('../../src/Schema');

describe('Schema.number', () => {
  test('Returns valid number', () => {
    const testData = { test: 2 };

    expect(Schema.number('test', testData)).toBe(testData.test);
  });

  test('Returns error', () => {
    const testData = { test: 'string' };

    const checked = Schema.number('test', testData);

    expect(checked).toBeInstanceOf(TypeError);
    expect(checked).toMatchSnapshot();
  });

  test('__type', () => {
    expect(Schema.number.__type).toBe('number');
  });
});
