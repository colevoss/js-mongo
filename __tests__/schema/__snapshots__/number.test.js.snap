// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Schema.number Returns error 1`] = `[TypeError: Expected test to be of type number but recieved string of string]`;
